File: web/src/api/research.ts
import { request } from './client';
import type { ResearchDetail, ResearchPage, ResearchRecord } from './types';

export const readQuestions = (): Promise<string[]> => request<string[]>('/research/questions');

export const readRecords = (): Promise<ResearchPage> => request<ResearchPage>('/research/records');

export const createRecord = (query: string): Promise<ResearchRecord> =>
  request<ResearchRecord>('/research/records', { method: 'POST', body: { query } });

export const readRecord = (recordId: string): Promise<ResearchDetail> =>
  request<ResearchDetail>(`/research/records/${recordId}`);

export const deleteRecord = (recordId: string): Promise<void> =>
  request<void>(`/research/records/${recordId}`, { method: 'DELETE' });

File: server/app/routers/research.py
"""企业背调接口。"""

from __future__ import annotations

from fastapi import APIRouter, Depends, HTTPException, status

from ..deps import current_user
from ..models import (
    CreateResearchRequest,
    ResearchDetail,
    ResearchPage,
    ResearchRecord,
    User,
)
from ..mock.catalog import RESEARCH_QUESTIONS
from ..repositories import research

router = APIRouter(prefix="/api/research", tags=["research"])


@router.get("/questions", response_model=list[str])
def read_questions() -> list[str]:
    return list(RESEARCH_QUESTIONS)


@router.get("/records", response_model=ResearchPage)
def read_records(user: User = Depends(current_user)) -> ResearchPage:
    return research.summary()


@router.post("/records", response_model=ResearchRecord, status_code=status.HTTP_201_CREATED)
def create_record(payload: CreateResearchRequest, user: User = Depends(current_user)) -> ResearchRecord:
    return research.create(payload.query)


@router.get("/records/{record_id}", response_model=ResearchDetail)
def read_record(record_id: str, user: User = Depends(current_user)) -> ResearchDetail:
    detail = research.detail(record_id)
    if detail is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="调研记录不存在")
    return detail


@router.delete("/records/{record_id}", status_code=status.HTTP_204_NO_CONTENT, response_model=None)
def delete_record(record_id: str, user: User = Depends(current_user)) -> None:
    if not research.delete(record_id):
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="调研记录不存在")
